### PR TITLE
Fixes #571: Maps feature for Susper

### LINKED ISSUE #571: Implement Maps feature to Susper
**Expected Behaviour**

Add a new feature Maps to Susper. We're creating Susper intelligent using SUSI AI. Susi uses openstreet maps API. We can implement maps using openstreet map.

Screenshot of OpenStreet Map: 

screenshot from 2017-06-29 01-11-57

### PR DESCRIPTION
**Pull request by harshit98**
Fixes issue #571 

Changes: 
- Added Maps feature.
- Added feature in which Map will tell about your place (where you are at present).

Demo Link: http://harshit98.github.io/susper.com

Screenshots for the change: 

Currently, it looks like this: 
screenshot from 2017-07-02 18-36-54

I'm facing issue=>overlap of the map with results. We will display map by a newly created tab "Maps".  I tried to create a separate component for the map to avoid overlapping with basic results but, it is not working as it should also provide a tab switching effect.

Kindly review @nikhilrayaprolu @Marauderer97

### CODE REVIEW

**Comment by Marauderer97** [comment]
@harshit98 I don't understand why creating it in a seperate tab was not possible? :worried:

**Comment by harshit98** [comment]
@Marauderer97 I read about Openstreet maps with nodejs. The map will only work, when it is being called inside `ngOnInit()` function.

**Comment by nikhilrayaprolu** [comment]
@harshit98 I will try to find some alternative to this.

File: package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "karma-jasmine": "^1.1.0",
     "karma-jasmine-html-reporter": "^0.2.2",
     "karma-remap-istanbul": "^0.6.0",
+    "leaflet": "^1.1.0",
```
Comment (nikhilrayaprolu) [inline package.json]: I think this should not be in devdependencies.

File: src/app/results/results.component.html [hunk 2]
```diff
@@ -67,6 +68,9 @@
 
   <app-infobox [hidden]="hidefooter" class="infobox col-md-4" *ngIf="Display('all')"></app-infobox>
   <!-- END -->
+  <div id="map">
```
Comment (nikhilrayaprolu) [inline src/app/results/results.component.html]: wrap this with `Display="maps"`

File: src/app/results/results.component.html [hunk 3]
```diff
@@ -67,6 +68,9 @@
...
+    <script src="https://unpkg.com/leaflet@1.1.0/dist/leaflet.js"></script>
```
Comment (nikhilrayaprolu) [inline src/app/results/results.component.html]: this is a bad practice to have script tags and link tags in components

Comment (harshit98) [inline src/app/results/results.component.html]: Leaflet map is called inside a div tag with `id="map"` and renders map if script is inside div tag. But it looks like leaflet is failing tests with typescript 2.0. We need to find an alternative to pass the tests.

**Pull request closed without merging**